src/overlays/actors/ovl_En_Fish/z_en_fish.c

```diff
@@ -47,73 +63,709 @@ static ColliderJntSphInit D_80A17040 = {
+        } else {
+            temp_a0_2 = (this->actor.yawTowardsPlayer + 0x7000);
+        }
+
+        sp38.x = player->actor.world.pos.x + (Math_SinS(temp_a0_2) * 20.0f);
+        sp38.y = player->actor.world.pos.y;
+        sp38.z = player->actor.world.pos.z + (Math_CosS(temp_a0_2) * 20.0f);
+
+        yaw = Math_Vec3f_Yaw(&this->actor.world.pos, &sp38);
+        Math_StepToAngleS(&this->actor.world.rot.y, yaw, 3000);
+    }
+
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    this->skelAnime.playSpeed = CLAMP_MAX((this->actor.speedXZ * 1.5f) + 0.8f, 4.0f);
+
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->timer <= 0) {
+        EnFish_Respawning_SetupSlowDown(this);
+    }
+}
+
+// Dropped type functions
+
+void EnFish_Dropped_SetupFall(EnFish* this) {
+    this->actor.gravity = -1.0f;
+    this->actor.minVelocityY = -10.0f;
+    this->actor.shape.yOffset = 0.0f;
+    EnFish_SetOutOfWaterAnimation(this);
+    this->unk_250 = 5;
+    this->actionFunc = EnFish_Dropped_Fall;
+    this->timer = 300;
+}
+
+void EnFish_Dropped_Fall(EnFish* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 0.1f, 0.1f, 0.0f);
+    Math_StepToAngleS(&this->actor.world.rot.x, 0x4000, 100);
+    Math_StepToAngleS(&this->actor.world.rot.z, -0x4000, 100);
+    this->actor.shape.rot.x = this->actor.world.rot.x;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    this->actor.shape.rot.z = this->actor.world.rot.z;
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->actor.bgCheckFlags & 1) { // On floor
+        this->timer = 400;
+        EnFish_Dropped_SetupFlopOnGround(this);
+    } else if (this->actor.bgCheckFlags & 0x20) { // In water
+        EnFish_Dropped_SetupSwimAway(this);
+    } else if ((this->timer <= 0) && (this->actor.params == FISH_DROPPED) && (this->actor.floorHeight < -31990.0f)) {
+        osSyncPrintf(VT_COL(YELLOW, BLACK));
+        // BG missing? Running Actor_delete
+        osSyncPrintf("BG 抜け？ Actor_delete します(%s %d)\n", "../z_en_sakana.c", 822);
+        osSyncPrintf(VT_RST);
+        Actor_Kill(&this->actor);
+    }
+}
+
+/**
+ * If the fish is on a floor, this function is looped back to by EnFish_Dropped_FlopOnGround to set a new flopping
+ * height and whether the sound should play again.
+ */
+void EnFish_Dropped_SetupFlopOnGround(EnFish* this) {
+    s32 pad;
+    f32 randomFloat;
+    s32 playSound;
+
+    this->actor.gravity = -1.0f;
+    this->actor.minVelocityY = -10.0f;
+    randomFloat = Rand_ZeroOne();
+
+    if (randomFloat < 0.1f) {
+        this->actor.velocity.y = (Rand_ZeroOne() * 3.0f) + 2.5f;
+        playSound = true;
+    } else if (randomFloat < 0.2f) {
+        this->actor.velocity.y = (Rand_ZeroOne() * 1.2f) + 0.2f;
+        playSound = true;
+    } else {
+        this->actor.velocity.y = 0.0f;
+
+        if (Rand_ZeroOne() < 0.2f) {
+            playSound = true;
+        } else {
+            playSound = false;
+        }
```

Comment:
I guess this doesn't work, right?
```suggestion
        playSound = Rand_ZeroOne() < 0.2f;
```

Reply:
It needs an extra branch. Can go with `playSound = (Rand_ZeroOne() < 0.2f ? true : false);` if that's better.